gslib/cloud_api_delegator.py

```diff
@@ -209,6 +210,13 @@ def GetApiSelector(self, provider=None):
     elif self.prefer_api in (
         self.api_map[ApiMapConstants.SUPPORT_MAP][selected_provider]):
       api = self.prefer_api
+
+    if (api == ApiSelector.XML and context_config.get_context_config() and
+        context_config.get_context_config().use_client_certificate):
+      self.logger.warn(
```

Comment:
I think we should raise a warning here? Since mTLS is a security feature, if a user is opting in for this feature and it is not going to work, they should be alerted to either make the proper API choice or set use_client_certificate to False.

Reply:
Done and created test file